normalize.js

```diff
@@ -332,41 +328,39 @@ var buildCustomSchema = exports.buildCustomSchema = function (schema, types, par
       case 'group':
       case 'global_field':
         var newparent = parent.concat('_', field.uid);
-        var result = buildCustomSchema(field.schema, types, newparent, prefix);
-        if (result) {
-          for (var key in result.fields) {
-            if (Object.prototype.hasOwnProperty.call(result.fields[key], 'type')) {
-              result.fields[key] = result.fields[key].type;
-            }
+        var result = buildCustomSchema(field.schema, types, references, groups, newparent, prefix);
+
+        for (var key in result.fields) {
+          if (Object.prototype.hasOwnProperty.call(result.fields[key], 'type')) {
+            result.fields[key] = result.fields[key].type;
           }
-          if ((0, _keys2.default)(result.fields).length > 0) {
-            var _type = 'type ' + newparent + ' ' + (0, _stringify2.default)(result.fields).replace(/"/g, '');
-            types.push(_type);
-            fields[field.uid] = {
-              resolve: function resolve(source) {
-                if (field.multiple && !Array.isArray(source[field.uid])) {
-                  return [];
-                }
-                return source[field.uid] || null;
-              }
-            };
-            if (field.mandatory) {
-              if (field.multiple) {
-                fields[field.uid].type = '[' + newparent + ']!';
-              } else {
-                fields[field.uid].type = newparent + '!';
-              }
-            } else if (field.multiple) {
-              fields[field.uid].type = '[' + newparent + ']';
+        }
+        if ((0, _keys2.default)(result.fields).length > 0) {
+          var _type = 'type ' + newparent + ' ' + (0, _stringify2.default)(result.fields).replace(/"/g, '');
+          types.push(_type);
+
+          groups.push({
+            parent: parent,
+            field: field
+          });
+
+          if (field.mandatory) {
+            if (field.multiple) {
+              fields[field.uid] = '[' + newparent + ']!';
             } else {
-              fields[field.uid].type = '' + newparent;
+              fields[field.uid] = newparent + '!';
             }
+          } else if (field.multiple) {
+            fields[field.uid] = '[' + newparent + ']';
+          } else {
+            fields[field.uid] = '' + newparent;
           }
         }
+
         break;
       case 'blocks':
         var blockparent = parent.concat('_', field.uid);
-        var blockType = buildBlockCustomSchema(field.blocks, types, blockparent, prefix);
+        var blockType = buildBlockCustomSchema(field.blocks, types, references, blockparent, prefix);
```

Comment:
why are the groups not being passed as a parameter here?

This leads to crashes during schema generation for our usecase

Reply:
Thanks for the notifying this. We have added the missing param.

Reply:
i can confirm this works for us.